org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java

```diff
@@ -0,0 +1,299 @@
+                model.eResource().getErrors());
+        
+        // Find all the Instantiations.
+        Instantiation a1 = null;
+        Instantiation a2 = null;
+        Instantiation b1 = null;
+        Instantiation b2 = null;
+        
+        
+        List<EObject> objs = asStream(model.eAllContents()).filter(obj -> 
+            (obj instanceof Instantiation)
+        ).collect(Collectors.toList());
+        for (EObject obj : objs) {
+            if (obj instanceof Instantiation) {
+                Instantiation inst = (Instantiation)obj;
+                if (inst.getName() == "a1") {
+                    a1 = inst;
+                }
+                if (inst.getName() == "a2") {
+                    a2 = inst;
+                }
+                if (inst.getName() == "b1") {
+                    b1 = inst;
+                }
+                if (inst.getName() == "b1") {
+                    b2 = inst;
+                }
+            }
+        }
+                
+        // Construct all relevant instantiation lists.
+        var list_a1 = new LinkedList<Instantiation>();
+        list_a1.add(a1);
+        var list_a2 = new LinkedList<Instantiation>();
+        list_a2.add(a2);
+        var list_a1b1 = new LinkedList<Instantiation>();
+        list_a1b1.add(a1);
+        list_a1b1.add(b1);
+        var list_a2b1 = new LinkedList<Instantiation>();
+        list_a2b1.add(a2);
+        list_a2b1.add(b1);
+        var list_a1b2 = new LinkedList<Instantiation>();
+        list_a1b2.add(a1);
+        list_a1b2.add(b2);
+        var list_a2b2 = new LinkedList<Instantiation>();
+        list_a2b2.add(a2);
+        list_a2b2.add(b2);
+        var list_b1 = new LinkedList<Instantiation>();
+        list_b1.add(b1);
+        var list_b2 = new LinkedList<Instantiation>();
+        list_b2.add(b2);
+        
+        /* Check for this:
+         *     initialValue(x, null) returns 1
+         *     initialValue(x, [a1]) returns 2
+         *     initialValue(x, [a2]) returns -1
+         *     initialValue(x, [a1, b1]) returns 3
+         *     initialValue(x, [a2, b1]) returns -1
+         *     initialValue(x, [a1, b2]) returns -2
+         *     initialValue(x, [a2, b2]) returns -1
+         * 
+         *     initialValue(y, null) returns 2
+         *     initialValue(y, [a1]) throws an IllegalArgumentException
+         *     initialValue(y, [b1]) returns 3
+         *     initialValue(y, [b2]) returns -2
+         */
+        
+        model.eAllContents().forEachRemaining((obj) -> {
+            if (obj instanceof Parameter) {
+                Parameter parameter = (Parameter)obj;
+                if (parameter.getName() == "x") {
+                    var values = ASTUtils.initialValue(parameter, null);
+                    Assertions.assertEquals(values.get(0).getLiteral(), "1");
+                    
+                    values = ASTUtils.initialValue(parameter, list_a1);
```

Comment:
```suggestion
                    values = ASTUtils.initialValue(parameter, List.of(a1));
```
I would suggest removing all those list variables and replacing them with `List.of` like this. Alternatively you could also use `List.of` in the definition of those variables, which would save lines compared to LinkedList. I don't think using variables here nets any readability though

Reply:
Thanks for the suggestion. Will do!